include/no_os_mdio.h

```diff
@@ -74,6 +74,8 @@ struct no_os_mdio_init_param {
 	struct no_os_mdio_ops *ops;
 	/** Platform or implementation specific parameters. */
 	void *extra;
+	/** MDIO mutex*/
+	void 		*mutex;
```

Comment:
I think there may be the case where the user declares more than 1 descriptor for a peripheral. By placing the mutex in here, it won't be shared between descriptors, which might cause issues if 2 descriptors are used in 2 different threads. One possible solution would be to declare it as `static` in the corresponding source files (`no_os_mdio.c` in this case).

Reply:
In general it is correct. However if we switch to `static` declaration it won't be possible to declare for instance two different SPI descriptors, that might have different ports and might be used simultaneously.

Reply:
Modified i2c and spi api to support bus descriptor, will add there